src/forms/inputcounter/InputCounter.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+import * as React from "react";
+import * as utils from "../../utils";
+import {Icon} from "../../icon";
+import {Input} from "../input/Input";
+import "./InputCounter.css";
+
+type InputCounterProps = {
+    className?: string;
+    maxValue?: number;
+    minValue?: number;
+    defaultValue?: number;
+    errMsg?: string;
+    width?: string;
+    onChange?: (evt: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+type InputCounterState = {
+    value: number;
+    errorMsg: string;
+};
+
+export class InputCounter extends React.PureComponent<InputCounterProps, InputCounterState> {
+    public static defaultProps: Partial<InputCounterProps> = {
+        minValue: Number.MIN_SAFE_INTEGER,
+        maxValue: Number.MAX_SAFE_INTEGER,
+        errMsg: "Value should be",
+        width: "6rem",
+    };
+    constructor(props: any) {
+        super(props);
+
+        //Default Value should be >= to minValue
+        let defaultValue = isNaN(props.defaultValue) ? 0 : Math.max(props.defaultValue, props.minValue);
+        //Default Value should be <= to minValue
+        defaultValue = Math.min(defaultValue, props.maxValue);
```

Comment:
What is purpose of changing defaultValue here again ?  I guess defaultValue is given by user and you just need to check if it is between Min and Max value. Can't we do this in one check? And what if user provide wrong default value ?

Reply:
DefaultValue should be >= MinValue and  <= MaxValue. So even if user gives the defaultValue < MinValue or defaultValue>maxValue then we need to adjust the defaultValue as per minValue and maxValue.
So to address the comment given by priya: "What happens if the defaultValue is greater that maxVal?" , I have added the code to select the min value between defaultValue and maxValue.

Reply:
@girishdudhe  Its not wise to change value without knowledge of user. We should give some error. I am not sure but please check if we  can add such conditions while declaration of props itself